### PR TITLE
lldb: Swift async: handle 2 different coroutine ABIs

### PR DESCRIPTION
**Pull request by fredriss**

### CODE REVIEW

**Comment by fredriss** [comment]
@swift-ci test

**Comment by fredriss** [comment]
@swift-ci test

File: lldb/source/Target/SwiftLanguageRuntime.cpp [hunk 1]
```diff
@@ -2426,9 +2426,10 @@ SwiftLanguageRuntime::GetRuntimeUnwindPlan(ProcessSP process_sp,
   // and thus it needs to be dereferenced to get to the context for the
   // function. The debug info for locals reflects this difference, so our
   // unwinding of the context reister needs to reflect it too.
-  bool in_resume_function =
-      sc.symbol->GetName().GetStringRef().contains(".resume.");
-  if (in_resume_function) {
+  bool indirect_context =
+      sc.symbol->GetName().GetStringRef().contains(
+          "await resume partial function");
+  if (indirect_context) {
```
Comment (kastiglione) [inline lldb/source/Target/SwiftLanguageRuntime.cpp]: In other files we have helper functions to do this kind of a check using swift demangling. The equivalent check would be to look for a `AsyncAwaitResumePartialFunction` child of the top level ("Global") node. I don't think it blocks this, if you prefer to commit as is, it should be a relatively simple NFC later.

Note the comment above this still references ".resume.".

**Comment by adrian-prantl** [comment]
This will cause a merge conflict, since the this includes an older version of the DWARFExpression patch.

The good news is we just need to wait until 

https://smooshbase.apple.com/swiftci/view/Auto-merger/view/llvm-project/job/fs-llvm-project-apple-stable-20210107-to-swift-main/187/

succeeds.

**Comment by fredriss** [comment]
@swift-ci test

**Comment by fredriss** [comment]
@adrian-prantl this has the cherry-pick of the new version of the DWARFExpression change
@kastiglione I added a helper to SwiftLanguageRuntimeNames to query the kind of funclet I'm interested in

**Pull request merged**